imports/client/components/PuzzlePage.tsx

```diff
@@ -110,116 +112,131 @@ interface ChatMessageProps {
+  });
+  const classes = classnames('chat-message', props.isSystemMessage && 'system-message');
+
+  return (
+    <div className={classes}>
+      {!props.suppressSender && <span className="chat-timestamp">{ts}</span>}
+      {!props.suppressSender && <strong>{props.senderDisplayName}</strong>}
+      <span
+        // eslint-disable-next-line react/no-danger
+        dangerouslySetInnerHTML={{ __html: marked(DOMPurify.sanitize(props.message.text)) }}
+      />
+    </div>
+  );
+});
 
 interface ChatHistoryProps {
   chatMessages: FilteredChatMessageType[];
   displayNames: Record<string, string>;
 }
 
-class ChatHistory extends React.PureComponent<ChatHistoryProps> {
-  ref: React.RefObject<HTMLDivElement>
-
-  resizeHandler?: () => void;
-
-  shouldScroll: boolean;
-
-  constructor(props: ChatHistoryProps) {
-    super(props);
-    this.ref = React.createRef();
-    this.shouldScroll = false;
-  }
-
-  componentDidMount() {
-    // Scroll to end of chat.
-    this.forceScrollBottom();
-
-    // Make sure when the window is resized, we stick to the bottom if we were there
-    this.resizeHandler = () => {
-      this.maybeForceScrollBottom();
-    };
-
-    window.addEventListener('resize', this.resizeHandler);
-  }
-
-  componentDidUpdate() {
-    this.maybeForceScrollBottom();
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.resizeHandler!);
-  }
+type ChatHistoryHandle = {
+  forceScrollBottom: () => void;
+  maybeForceScrollBottom: () => void;
+}
 
-  onScroll = () => {
-    this.saveShouldScroll();
-  };
+// TODO: chat scrolling is still kinda broken -- on initial pageload, it seems that we render, then only after the viewers data loads,
+// we draw boxes for viewers, which results in ChatHistory getting placed in a smaller area, but it doesn't get any event/callback for this.
+// The solution is probably to make sure that when ChatPeople rerenders, we find some way to trigger a maybeForceScrollBottom() here.
+const ChatHistory = React.forwardRef((props: ChatHistoryProps, forwardedRef: React.Ref<ChatHistoryHandle>) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const shouldScroll = useRef<boolean>(true);
 
-  saveShouldScroll = () => {
+  const saveShouldScroll = useCallback(() => {
     // Save whether the current scrollTop is equal to the ~maximum scrollTop.
     // If so, then we should make the log "stick" to the bottom, by manually scrolling to the bottom
     // when needed.
-    const messagePane = this.ref.current!;
+    const messagePane = ref.current;
+    if (!messagePane) {
+      return;
+    }
 
     // Include a 5 px fudge factor to account for bad scrolling and
     // fractional pixels
-    this.shouldScroll = (messagePane.clientHeight + messagePane.scrollTop + 5 >= messagePane.scrollHeight);
-  };
+    shouldScroll.current = (messagePane.clientHeight + messagePane.scrollTop + 5 >= messagePane.scrollHeight);
+    console.log(`shouldScroll: ${shouldScroll.current}`);
```

Comment:
Is this debugging cruft? (Why didn't eslint complain?)

Reply:
Yeah.  I probably should have taken it out for this PR, though I likely still need it to look into a proper fix for #427.

eslint didn't complain because we disable the `max-len` and `no-console` rules at the top of this file.